test_simple_graph.py

```diff
@@ -75,3 +128,15 @@ def test_adjacent(populated):
 def test_adjacent_none(populated):
     with pytest.raises(KeyError):
         populated.adjacent(58, 28)
+
+
+def test_depth_first(cyclic):
+    nodes = cyclic.depth_first_traversal(1)
```

Comment:
This doesn't test that your traversal is in the correct order, which is kind of the whole point of a traversal. You should work out the correct order on paper and then test it against that.